framework/src/governance/lib/datazone/datazone-custom-asset-type-factory.ts

```diff
@@ -0,0 +1,189 @@
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: DataZoneCustomAssetTypeFactory.name,
+    };
+
+    super(scope, id, trackedConstructProps);
+
+    const stack = Stack.of(this);
+
+    this.domainId = props.domainId;
+
+    this.removalPolicy = Context.revertRemovalPolicy(scope, props.removalPolicy);
+
+    this.handlerRole = new Role(this, 'HandlerRole', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName('service-role/AWSLambdaBasicExecutionRole'),
+      ],
+      inlinePolicies: {
+        DataZonePermission: new PolicyDocument({
+          statements: [
+            new PolicyStatement({
+              effect: Effect.ALLOW,
+              actions: [
+                'datazone:CreateFormType',
+                'datazone:CreateAssetType',
+                'datazone:DeleteAssetType',
+                'datazone:DeleteFormType',
+                'datazone:GetFormType',
+              ],
+              resources: [`arn:${stack.partition}:datazone:${stack.region}:${stack.account}:domain/${this.domainId}`],
+            }),
+          ],
+        }),
+      },
+    });
+
+    const provider = new DsfProvider(this, 'Provider', {
+      providerName: 'DataZoneCustomAssetType',
+      onEventHandlerDefinition: {
+        depsLockFilePath: __dirname+'/resources/datazone-custom-asset-type/package-lock.json',
+        entryFile: __dirname+'/resources/datazone-custom-asset-type/index.mjs',
+        handler: 'index.handler',
+        iamRole: this.handlerRole,
+        timeout: Duration.minutes(2),
+      },
+      removalPolicy: this.removalPolicy,
+    });
+
+    this.createLogGroup = provider.onEventHandlerLogGroup;
+    this.createFunction = provider.onEventHandlerFunction;
+    this.createRole = provider.onEventHandlerRole;
+    this.serviceToken = provider.serviceToken;
+  }
+
+  /**
+   * Creates a DataZone custom asset type based on the provided properties
+   * @param id the ID of the CDK Construct
+   * @param customAssetType the properties of the custom asset type
+   * @returns the custom asset type
+   */
+  public createCustomAssetType(id: string, customAssetType: DataZoneCustomAssetTypeProps): CustomAssetType {
+
+    // create a project membership for the custom resource role so it can create custom asset types in this project
+    const projMembership = new CfnProjectMembership(this, `${id}ProjectMembership`, {
+      designation: 'PROJECT_OWNER',
+      domainIdentifier: this.domainId,
+      member: {
+        userIdentifier: this.handlerRole.roleArn,
+      },
+      projectIdentifier: customAssetType.projectId,
+    });
+
+    // The custom resource creating the custom asset type
+    const crResp = new CustomResource(this, id, {
+      serviceToken: this.serviceToken,
+      removalPolicy: this.removalPolicy,
+      properties: {
```

Comment:
Why use the construct level removalPolicy? Are there senarios where we want to keep resource in case of deleting the CR?

Reply:
if we want to keep the subscription grants in DataZone